### PR TITLE
Update namespace docs

### PR DESCRIPTION
**Pull request by Tratcher**
Hosting and a few misc namespaces.

Contributes to dotnet/aspnetcore#27286.

### CODE REVIEW

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Hosting.Server.Abstractions.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Hosting.Server.Abstractions">
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Provides the <see cref="T:Microsoft.AspNetCore.Hosting.Server.Abstractions.IHostContextContainer[T]" /> type that can be implemented by a server to pool <see cref="T:Microsoft.AspNetCore.Http.HttpContext" /> instances.</summary>
```
Comment (Tratcher) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Hosting.Server.Abstractions.xml:3]: What's the proper type syntax for generics?

Comment (Rick-Anderson) [reply aspnet-core/xml/ns-Microsoft.AspNetCore.Hosting.Server.Abstractions.xml:3]: I believe it's 
`<xref:Microsoft.AspNetCore.Hosting.Server.Abstractions.IHostContextContainer%601>` or
` <xref:Microsoft.AspNetCore.Hosting.Server.Abstractions.IHostContextContainer%601?displayProperty=nameWithType>`

to display name with type.

File: aspnet-core/xml/ns-Microsoft.AspNetCore.ResponseCaching.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.ResponseCaching">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types used to help cache responses and improve site performance.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.ResponseCaching.xml:3]:
```suggestion
    <summary>Contains types used to cache responses and improve site performance.</summary>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.ResponseCompression.xml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.ResponseCompression">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types used to dynamically compress response bodies and save bandwidth.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.ResponseCompression.xml:3]:
```suggestion
    <summary>Contains types used to dynamically compress response bodies and reduce bandwidth.</summary>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Rewrite.xml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <Namespace Name="Microsoft.AspNetCore.Rewrite">
   <Docs>
-    <summary>To be added.</summary>
-    <remarks>To be added.</remarks>
+    <summary>Contains types use to modify request attributes based on predefined rules. Rewriting can be valuable in cases where resources have moved, or may be accessed by multiple urls.</summary>
```
Comment (Rick-Anderson) [inline aspnet-core/xml/ns-Microsoft.AspNetCore.Rewrite.xml:3]:
```suggestion
    <summary>Contains types used to modify request attributes based on predefined rules. Rewriting can be valuable in cases where resources have moved, or may be accessed by multiple urls.</summary>
```

**Review by Rick-Anderson — APPROVED** [review]
Opinionated nits, your version is fine. Feel free to ignore.

File: aspnet-core/xml/ns-Microsoft.AspNetCore.Hosting.Server.Abstractions.xml (hunk 1 above)
Comment (Tratcher) [reply aspnet-core/xml/ns-Microsoft.AspNetCore.Hosting.Server.Abstractions.xml:3]:
```suggestion
    <summary>Provides the <see cref="T:Microsoft.AspNetCore.Hosting.Server.Abstractions.IHostContextContainer%601" /> type that can be implemented by a server to pool <see cref="T:Microsoft.AspNetCore.Http.HttpContext" /> instances.</summary>
```

File: aspnet-core/xml/ns-Microsoft.AspNetCore.ResponseCompression.xml (hunk 3 above)
Comment (Tratcher) [reply aspnet-core/xml/ns-Microsoft.AspNetCore.ResponseCompression.xml:3]: That's a bit confusing, you're not reducing the bandwidth, you're reducing the bandwidth utilization.
```suggestion
    <summary>Contains types used to dynamically compress response bodies and reduce bandwidth utilization.</summary>
```

**Pull request merged**